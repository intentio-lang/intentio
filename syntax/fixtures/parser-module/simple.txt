fun foo(a, b) {
    a;
}

fun bar() {
    x = 4;
}
