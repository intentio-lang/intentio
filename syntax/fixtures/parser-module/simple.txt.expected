{
  "_moduleSourceItems": [
    {
      "_funDeclBody": [
        {
          "contents": {
            "contents": "a",
            "tag": "Id"
          },
          "tag": "IdExpr"
        }
      ],
      "_funDeclParams": [
        "a",
        "b"
      ],
      "_itemDeclName": "foo"
    },
    {
      "_funDeclBody": [
        {
          "_letDeclName": "x",
          "_letDeclVal": {
            "contents": {
              "contents": "4",
              "tag": "Integer"
            },
            "tag": "LitExpr"
          },
          "tag": "LetDeclExpr"
        }
      ],
      "_funDeclParams": [],
      "_itemDeclName": "bar"
    }
  ],
  "_moduleSourceName": "test"
}
